tox.ini

```diff
@@ -44,13 +44,32 @@ usedevelop = False
 # We use extras=test to get things like "mock" that are required for our unit
 # tests.
 extras = test
-commands =
-         trial {env:TAHOE_LAFS_TRIAL_ARGS:--rterrors} {posargs:allmydata}
-         tahoe --version
 
-[testenv:py36]
+setenv =
+       # Define TEST_SUITE in the environment as an aid to constructing the
+       # correct test command below.
+       !py36: TEST_SUITE = allmydata
+       py36: TEST_SUITE = allmydata.test.python3_tests
+
 commands =
-         trial {env:TAHOE_LAFS_TRIAL_ARGS:--rterrors} {posargs:allmydata.test.python3_tests}
+	 # As an aid to debugging, dump all of the Python packages and their
+	 # versions that are installed in the test environment.  This is
+	 # particularly useful to get from CI runs - though hopefully the
+	 # version pinning we do limits the variability of this output
+         pip freeze
+
+	 # The tahoe script isn't sufficiently ported for this to succeed on
+	 # Python 3.x yet.
+         !py36: tahoe --version
+
+         !coverage: trial {env:TAHOE_LAFS_TRIAL_ARGS:--rterrors} {posargs:{env:TEST_SUITE}}
+
+	 # measuring coverage is somewhat slower than not measuring coverage
+	 # so only do it on request.
+	 coverage: coverage run -m twisted.trial {env:TAHOE_LAFS_TRIAL_ARGS:--rterrors --reporter=timing} {posargs:{env:TEST_SUITE}}
+	 coverage: coverage combine
+         coverage: coverage xml
```

Comment:
Indentation looks a bit odd here, but then indentation looks a bit odd throughout this file. :-)

Reply:
Uh yea.  There were a bunch of tabs in this file.  I replaced them all with spaces.